UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentAncillaryFiles .cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Globalization;
+using System.IO;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.Common.Configuration;
+using UKHO.ExchangeSetService.Common.Helpers;
+using UKHO.ExchangeSetService.Common.Logging;
+
+namespace UKHO.ExchangeSetService.FulfilmentService.Services
+{
+    public class FulfilmentAncillaryFiles : IFulfilmentAncillaryFiles
+    {
+        private readonly IOptions<FileShareServiceConfiguration> fileShareServiceConfig;
+        private readonly IFileSystemHelper fileSystemHelper;
+        private readonly ILogger<FulfilmentDataService> logger;
+
+        public FulfilmentAncillaryFiles(IOptions<FileShareServiceConfiguration> fileShareServiceConfig, IFileSystemHelper fileSystemHelper, ILogger<FulfilmentDataService> logger)                                       
+        {
+            this.fileShareServiceConfig = fileShareServiceConfig;
+            this.fileSystemHelper = fileSystemHelper;
+            this.logger = logger;
+        }
+
+        public async Task<bool> CreateSerialEncFile(string batchId, string exchangeSetPath, string correlationId)
+        {
+            if (!string.IsNullOrWhiteSpace(exchangeSetPath))
+            {
+                string serialFilePath = Path.Combine(exchangeSetPath, fileShareServiceConfig.Value.SerialFileName);
+                fileSystemHelper.CheckAndCreateFolder(exchangeSetPath);
+                CultureInfo cultureInfo = CultureInfo.InvariantCulture;
+                int weekNumber = cultureInfo.Calendar.GetWeekOfYear(DateTime.UtcNow, CalendarWeekRule.FirstFourDayWeek, DayOfWeek.Monday);
```

Comment:
I'm not sure how often it would be used but I would be tempted to extract these two lines into a helper file which takes a Date and returns the Week Number. That would enable this bit of the code to be tested i.e. to make sure we do get back the correct week number.

It might also be useful in other parts of the code - I don't know but I'm guessing the week number may be required elsewhere?

Reply:
I think this may return the wrong week number at the beginning of the week.  The UKHO week number starts on the 1st Thursday of January.  i.e. 2021/1/7 is the beginning of week 2021/1,  2021/1/6 was the end of week 2020/52.

I believe you need:
```suggestion
                int weekNumber = cultureInfo.Calendar.GetWeekOfYear(DateTime.UtcNow, CalendarWeekRule.FirstFullWeek, DayOfWeek.Thursday);
```

Reply:
done the suggested changes